states/huntsman/coarse_focusing.py

```diff
@@ -0,0 +1,29 @@
+"""
+State to handle the coarse focusing at the start of the night.
+"""
+from pocs.utils import current_time
+
+
+def on_enter(event_data):
+    '''
+    Coarse focusing state. Will do a coarse focus for each camera and move to
+    the scheduling state.
+    '''
+    pocs = event_data.model
+    pocs.next_state = 'parking'
+
+    coarse_focus_timeout = pocs.config['focusing']['coarse']['timeout']
+    try:
+        # Do the autofocusing
+        pocs.say("Coarse focusing all cameras before starting observing for the night.")
+        autofocus_events = pocs.observatory.autofocus_cameras(coarse=True)
+        pocs.logger.debug("Waiting for coarse focus to finish.")
+        pocs.wait_for_events(list(autofocus_events.values()), coarse_focus_timeout)
+
+        # Update last focus time
+        pocs.observatory.last_focus_time = current_time()
```

Comment:
I still think this is the wrong criterion. It doesn't matter if it's been 6 or however many hours since the last coarse focus provided that the focus has been continually tweaked in the meantime with fine focusing at the start of every dither sequence. Should be using the time of the last completed fine focus here, not coarse focus, and require a fine focus to have taken place in the last 2 hours, say (they normally happen every ~1 hour while observing).

Can update the `last_focus_time`at the end of the `Observatory.autofocus_cameras()` method.